### ISSUE TITLE
Using Lua libraries rarely works due to strict Lua semantics used in Terra

### ISSUE THREAD

**Issue by renatoathaydes** [terralang/terra#570]
I have been trying to use Lua testing tools, like LuaUnit and busted, but neither works in Terra.

The reason seems related to enforcement of declaring variables before use.

Example error trying to run busted:

```
/usr/local/share/lua/5.1/pl/compat.lua:228: Global variable 'warn' is not declared. Global variables must be 'declared' through a regular assignment (even to nil) at global scope before being used.
stack traceback:
	[C]: in function 'error'
	/Users/runner/work/terra/terra/src/strict.lua:39: in function '__index'
	/usr/local/share/lua/5.1/pl/compat.lua:228: in main chunk
	[C]: in function 'require'
	/usr/local/share/lua/5.1/pl/utils.lua:9: in main chunk
	[C]: in function 'require'
	/usr/local/share/lua/5.1/pl/path.lua:21: in main chunk
	[C]: in function 'require'
	/usr/local/share/lua/5.1/busted/runner.lua:3: in main chunk
	[C]: in function 'require'
	testing.lua:1: in main chunk
```

Both frameworks work fine in Lua, of course.

Is it possible to use Lua standard semantics in the Lua code when "building" Terra code?

**Comment by renatoathaydes** [comment]
BTW I use Lua 5.4, and just symlinked the libs to the 5.1 directory... I think the error is unrelated to that, but would it be possible to "upgrade" Lua to 5.4 or the problem is LuaJIT is still on 5.1?

**Comment by renatoathaydes** [comment]
Thanks for the information. I am a completely noobie in Lua and don't really care about Lua 5.1 VS 5.2, I just thought that, like in most other languages, the latest version if where everyone tries to stay at.

Anyway, that "busted" library does lots of funky stuff and has many deps, so I moved back to "luaunit" which has no dependencies and does the job... it works after I removed the metatable as you showed.
